### ISSUE TITLE
git_log() hides commits to branches before merge

### ISSUE THREAD

**Issue by lorenzwalthert** [r-lib/gert#99]
It seems only merge commits and commits made directly to the branch are shown in version 1.0.1 - those merged into the current branch via a merge are omitted. Here is the log of https://github.com/r-lib/styler via the git RStudio panel and `gert::git_log()`. Is there a way to also show these commits?

![Screenshot 2020-10-22 at 13 35 25](https://user-images.githubusercontent.com/10477073/96866638-defb4f00-146b-11eb-82fc-583496c527be.png)
![Screenshot 2020-10-22 at 13 35 11](https://user-images.githubusercontent.com/10477073/96866644-e1f63f80-146b-11eb-8518-5ccccee888cf.png)

**Comment by jeroen** [comment]
That's a good idea. Right now we only show the main branch log, indeed, and not the merged branches.

**Comment by lorenzwalthert** [comment]
Yes. I think think by default the command should show what its cli equivalent shows. Because this is unexpected for me and I am not sure I understand the rationale behind it.

**Comment by jeroen** [comment]
Will look into this for the next version, but if you need a quick solution right now, below an example how to lookup the commits from a merged branch:

```r
git_merged_branch_log <- function(commit){
  parents <- git_commit_info(commit)$parents
  if(length(parents) == 1)
    return(NULL)
  start_at <- git_merge_find_base(parents[1], parents[2])
  sublog <- git_log(parents[2])
  len <- which(sublog$commit == start_at)
  utils::head(sublog, len-1)
}

# Show merge branches for 10 latest commits
main <- git_log()
lapply(main$commit[1:10], git_merged_branch_log)
```

**Comment by lorenzwalthert** [comment]
Thanks @jeroen. What I am after finally is the equivalent of `$ git log`, but I just need the hashes, so I already resolved this with `system2('git', c('rev-list', 'HEAD'), stdout = TRUE)`. I know that's probably not the nice way but until the next release, I'll use this. Thanks again.
